### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by Nkaleth**
Learning objectives of this PR:

- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write integration tests with a React testing library.
- Use styles in a React app.
- Use React life cycle methods.
- Apply React best practices and language style guides in code.
- Use store, actions and reducers in React.

Video presentation of the project : https://www.loom.com/share/ad945539882e4738a1b85e593368239d

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -245,44 +189,21 @@ Feel free to check the [issues page](../../issues/).
...
-> Add at least 2 questions new developers would ask when they decide to use your project.
-
-- **[Question_1]**
-
-  - [Answer_1]
-
-- **[Question_2]**
-
-  - [Answer_2]
+I would like to thank Microverse<br>
```
Comment (jerryowusu) [inline README.md]: As part of the project requirements, your readme file should contain an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).
Kindly update to reflect this requirement. :+1:

File: src/styles/book.css [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+.weeksList {
+  padding: 2vh;
+  font-weight: 300;
+  font-size: 2vh;
+}
+
+.titleAuthorBook h2 {
+  margin-bottom: 0;
+}
```
Comment (jerryowusu) [inline src/styles/book.css]: As part of the project requirements, kindly make sure that your design follows the design guidelines in terms of the layout (composition and space between elements). There are some slight differences I see between your design and that of the actual design as shown below: 

- [x] Your design is missing the  small white arrow in a circle to the right side of the details card. 
- [x] The background of the individual details card is light  whilst the next one is darker and it follows that order downwards as shown below :point_down: 

Kindly check and update your design accordingly. 

### Your design 

Screenshot from 2023-03-16 23-53-03


### Actual design 


Screenshot from 2023-03-16 23-55-26

File: src/styles/booklist.css [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  background: #3a5b99;
+}
+
+.loading {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100vh;
+}
```
Comment (jerryowusu) [inline src/styles/booklist.css]: As part of the project requirements, kindly make sure that your design follows the design guidelines in terms of the layout (composition and space between elements). There are some slight differences I see between your design and that of the actual design as shown below: 

- [x] The position of the image covers the top part with the white small arrow in a circle at the top right corner with the text below them
- [x] The chequered nature of the books' background also follows a certain pattern. The first row starts with light background followed by darker background whiles the second row starts with dark background followed by lighter background. Each column also alternates in the same order as shown below. :point_down: 


### Your design 

Screenshot from 2023-03-16 23-54-59



### Actual design 

Screenshot from 2023-03-16 23-55-17

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @Nkaleth , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

almost-there-reach


## Highlights
Great job so far :heavy_check_mark: 
No linter errors. :heavy_check_mark: 
Nice commit messages :heavy_check_mark: 

## Required Changes ♻️

- [ ] As part of the project requirements, you are suppose to write unit tests for some pure functions in Redux code. However, all the test I see is unit tests for the components such as booklist and the categorylist. Kindly ensure to add unit tests for some of the pure functions in redux code. :pray: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/__snapshots__/categories.test.js.snap [hunk 4]
```diff
@@ -0,0 +1,209 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Testing Home Page: <CategoriesList /> component Should match the snapshot 1`] = `
+Array [
```
Comment (tufoinnkuo10) [inline src/__test__/__snapshots__/categories.test.js.snap]: - [ ] great job with the test. Please, like the previous reviewer noted, kindly add unit tests for some pure functions in Redux code. However, all the test I see is unit tests for the components such as booklist and the categorylist. Kindly ensure to add unit tests for some of the pure functions in Redux code. 🙏

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Nkaleth 

Great job making most of the changes. There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ Design on point
✅ Readme ok


-- [optional] please always check your project requirements. you were required to use git-flow on this milestone. see more [here](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md)

Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by pinkmoon25 — APPROVED** [review]
Hi @Nkaleth , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Linter checks :heavy_check_mark: 
gitflow :heavy_check_mark: 
Professional Readme :heavy_check_mark: 
Descriptive PR and commits :heavy_check_mark: 
React Redux :heavy_check_mark: 
Testing :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**